jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricherTest.java

```diff
@@ -44,24 +44,26 @@
     private ProcessorConfig processorConfig;
     private KubernetesListBuilder klb;
 
-    public void setUpExpectations() {
+    public void setUpExpectations(boolean shouldAddMockServiceToBuilder) {
         properties = new Properties();
         processorConfig = new ProcessorConfig();
         klb = new KubernetesListBuilder();
         // @formatter:off
-        klb.addToItems(getMockServiceBuilder()
-            .build());
         new Expectations() {{
             context.getProperties(); result = properties;
             context.getConfiguration().getProcessorConfig(); result = processorConfig;
         }};
         // @formatter:on
+        if (shouldAddMockServiceToBuilder) {
+            klb.addToItems(getMockServiceBuilder()
+                    .build());
+        }
```

Comment:
This is starting to look very confusing.

Try to refactor into things that are more understandable.

The method name gives no clue of what you're doing (it made sense when it was a `@Before `clause and all methods depended on it, now it doesn't).

If you are adding something to the list builder, then add it in the tests where is applicable, or use a different KLB where it's not:
```
// Option 1
// Given
setUpThingiesWithEmptyKLB();
klb.addToTiems(getMockServiceBuilder().build());

// vs Option 2
// Given
setupThingiesIncludingDefaultKLB();
final KubernetesListBuilder specificKLB = new KubernetesListBuilder();
...
```

The point of tests, besides verifying the code, is to understand what the code does with a quick glimpse.